src/ai/backend/manager/services/metric/container_metric.py

```diff
@@ -0,0 +1,219 @@
+
+    async def query_metadata(
+        self,
+        action: ContainerMetricMetadataAction,
+    ) -> ContainerMetricMetadataActionResult:
+        result = await self._query_label_values(CONTAINER_UTILIZATION_METRIC_LABEL_NAME)
+        return ContainerMetricMetadataActionResult(result["data"])
+
+    def _get_label_values_for_query(
+        self, label: ContainerMetricOptionalLabel, metric_name: str
+    ) -> list[str]:
+        label_values: list[str] = [
+            f'container_metric_name="{metric_name}"',
+        ]
+
+        def _append_if_not_none(value: Any, name: str) -> None:
+            if value is not None:
+                label_values.append(f'{name}="{value}"')
+
+        _append_if_not_none(label.value_type, "value_type")
+        _append_if_not_none(label.agent_id, "agent_id")
+        _append_if_not_none(label.kernel_id, "kernel_id")
+        _append_if_not_none(label.session_id, "session_id")
+        _append_if_not_none(label.user_id, "user_id")
+        _append_if_not_none(label.project_id, "project_id")
+        return label_values
+
+    def _get_sum_by_for_query(self, label: ContainerMetricOptionalLabel) -> list[str]:
+        sum_by_values = ["value_type"]
+
+        def _append_if_not_none(value: Any, name: str) -> None:
+            if value is not None:
+                sum_by_values.append(name)
+
+        _append_if_not_none(label.agent_id, "agent_id")
+        _append_if_not_none(label.kernel_id, "kernel_id")
+        _append_if_not_none(label.session_id, "session_id")
+        _append_if_not_none(label.user_id, "user_id")
+        _append_if_not_none(label.project_id, "project_id")
+        return sum_by_values
+
+    def _get_metric_type(
+        self,
+        metric_name: str,
+        label: ContainerMetricOptionalLabel,
+    ) -> UtilizationMetricType:
+        # TODO: Define device metadata for each metric and use it rather than hardcoding
+        match metric_name:
+            case "cpu_util" if label.value_type == "current":
+                return UtilizationMetricType.DIFF
+            case "net_rx" | "net_tx":
+                return UtilizationMetricType.RATE
+            case _:
+                return UtilizationMetricType.GAUGE
+
+    def _parse_query_string_by_metric_spec(
+        self,
+        param: MetricSpecForQuery,
+    ) -> str:
+        match param.metric_type:
+            case UtilizationMetricType.GAUGE:
+                return f"{param.str_sum_by()}(backendai_container_utilization{param.str_labels()})"
+            case UtilizationMetricType.RATE:
+                return (
+                    f"{param.str_sum_by()}(rate(backendai_container_utilization{param.str_labels()}[{param.timewindow}])) "
+                    f"/ {UTILIZATION_METRIC_INTERVAL}"
+                )
+            case UtilizationMetricType.DIFF:
+                return f"{param.str_sum_by()}(rate(backendai_container_utilization{param.str_labels()}[{param.timewindow}]))"
```

Comment:
Check this.
In DIFF, it looks good as increase, and I don't know if you should divide by interval in RATE.